libglusterfs/src/inode.c

```diff
@@ -831,10 +839,22 @@ inode_grep(inode_table_t *table, inode_t *parent, const char *name)
     pthread_mutex_lock(&table->lock);
     {
         dentry = __dentry_grep(table, parent, name, hash);
-        if (dentry) {
+
+        if (!dentry) {
+            pthread_mutex_unlock(&table->lock);
+            gf_msg_callingfn(xl->name, GF_LOG_WARNING, EINVAL,
+                             LG_MSG_INVALID_ARG,
+                             "Dentry grep failed for table: %s", table->name);
+
+            return NULL;
+        }
+
+        if (dentry->inode) {
             inode = dentry->inode;
-            if (inode)
-                __inode_ref(inode, false);
+        }
+
+        if (!__is_root_gfid(inode->gfid)) {
+            __inode_ref(inode, false, xl);
```

Comment:
This must be done only inside the `if` of line 852. Otherwise `inode` can be `NULL`.